lib/plugins/aws/lib/naming.js

```diff
@@ -169,7 +170,38 @@ module.exports = {
     }
     return `${this.provider.getStage()}-${this.provider.serverless.service.service}`;
   },
-  generateApiGatewayDeploymentLogicalId() {
+  hashString(resource) {
+    return crypto
+      .createHash('md5')
+      .update(resource)
+      .digest('hex');
+  },
+  getApiGatewayDeploymentLogicalId() {
+    let hashPostfix = '';
+    if (this.provider.serverless.service.provider.apiName &&
+      _.isString(this.provider.serverless.service.provider.apiName)) {
+      hashPostfix = this.hashString(this.provider.serverless.service.provider.apiName);
+    }
+    if (this.provider.serverless.service.provider.apiGateway &&
+        !_.isEmpty(this.provider.serverless.service.provider.apiGateway)) {
+      hashPostfix = this.hashString(hashPostfix +
+        JSON.stringify(this.provider.serverless.service.provider.apiGateway));
+    }
```

Comment:
I see that you're checking for the `apiGateway` and `apiName` properties for hashing, but what about `apiKeys`? shouldn't that also trigger a deployment?

Reply:
Correct me if I'm wrong, but from what I saw and tested `apiKeys` have depends on relationship with deployment. As in `usagePlan` that references `apiKey` `dependsOn` deployment and not other way around. At least that's how generated CloudFormation update looks like. That's why I didn't add it in hash calculations.

Reply:
hmmm right, but if you only make apiKeys or usagePlans changes, then API Keys & Usage Plans resources would update, but no deployment of such change would occur to apply those changes. Those updated resources would still `DependsOn` on the old deployment, which was not changed. Does that make sense?

I suggest we include them in the hash anyway just to be safe, cause the consequences are critical.

Reply:
Ok, added `apiKeys` to hash calculations